api/src/pcapi/core/operations/api.py

```diff
@@ -127,6 +130,28 @@ def get_last_date_for_event() -> datetime | None:
         )
 
 
+def _get_user_for_form(form: typeform.TypeformResponse) -> users_models.User | None:
+    user = None
+    if form.email:
+        user = users_models.User.query.filter(users_models.User.email == form.email).one_or_none()
+    if form.phone_number and not user:
+        try:
+            parsed_phone_number = ParsedPhoneNumber(form.phone_number)
+        except InvalidPhoneNumber:
+            pass
+        else:
+            users = (
+                users_models.User.query.filter(
+                    users_models.User._phoneNumber == parsed_phone_number.phone_number,
+                )
+                .limit(2)
+                .all()
+            )
+            if len(users) == 1:
+                user = users[0]
```

Comment:
```suggestion
            from contextlib import suppress
            # [...]
            with suppress(sqlalchemy.exc.MultipleResultsFound):
                user = (
                    users_models.User.query.filter(
                        users_models.User._phoneNumber == parsed_phone_number.phone_number,
                ).one_or_none()
```

Qu'en dis-tu ? C'est vraiment du détail ceci dit.